File: app/src/main/java/com/yobo/yobo_kotlin/test34_JiCheng/TestMain.java
package com.yobo.yobo_kotlin.test34_JiCheng;

/**
 * Created by ZhangBoshi
 * on 2019-12-30
 */


abstract class TestFather {
    public abstract void funA();
}
interface Test1 {
    void funB();

}
interface Test2 {
    void funC();
}

public class TestMain extends TestFather implements Test1,Test2{
    @Override
    public void funA() {

    }

    @Override
    public void funB() {

    }

    @Override
    public void funC() {

    }
}

File: app/src/main/java/com/yobo/yobo_kotlin/test1_26/test1/test23_inline/TestMain.kt
package com.yobo.yobo_kotlin.test1_26.test1.test23_inline

/**
 * Created by ZhangBoshi
 * on 2019-12-23
 */

//
//fun test1() {
//    val msg = "我是内联函数"
//    println(msg)
//}



inline fun test2(a: () -> String, noinline b: () -> Int) {
    val resultA = a.invoke()
    val resultB = b.invoke()
    println(resultA)
    println(resultB)

}
fun main() {

    test2({
        val name = "lambda闭包函数A"
        name

    }, {
        val age = 18
        age
    })
}